src/asciidoc/docs/guidelines/frontendUploading.adoc

```diff
@@ -0,0 +1,108 @@
+= Uploading files from the frontend
+:page-slug: frontend-uploading
+:page-description: How to enable users to upload files to the CDN from a Koji's frontend.
+
+Some templates provide users with the ability to upload files from a Koji’s frontend.
+For example, a public image board.
+When developing such a template, the recommended way to upload files is to use the `generateSignedUploadRequest` method from the <<withkoji-database-package#.generateSignedUploadRequest, @withkoji/database-sdk>> package.
+
+This method enables the file to be uploaded directly from the Koji’s frontend to the Koji CDN, for optimal performance.
+Other methods, such as `uploadFile`, would require the file to be uploaded to the project’s backend server and then to the CDN.
+
+To upload files with `generateSignedUploadRequest`, you configure the backend to acquire a presigned URL, which grants temporary access for uploading a file.
+Then, you configure the frontend to retrieve a presigned URL from the backend and use it to upload a file.
+
+== Prerequisites
+
+This guide covers advanced topics.
+It is suggested that you are familiar with the following topics before proceeding.
+
+* React and ES6 basics.
+* The Koji editor, <<instant-remixing#, remix process>>, and the <<withkoji-vcc-package,@withkoji/vcc>> package.
+* https://expressjs.com/[Express.js] web framework.
+* Backend communication and the <<withkoji-database-package,@withkoji/database-sdk>> package.
+* Performing HTTP requests (this example uses https://github.com/axios/axios[axios]).
+
+== Configuring the backend
+
+To upload from the frontend, you first need to acquire a presigned URL by using the Koji database SDK.
+
+In a backend service for your Koji project, set up an Express.js server with a POST method to the `/get-upload-url` endpoint.
+From this endpoint, call the following function.
+
+[source, javascript]
+----
+import Database from '@withkoji/database';
+
+const uploadImage = async (req, res) => {
```

Comment:
Might be good to change this function to be called `getUploadURL` or something similar?